src/core/compact_object.cc

```diff
@@ -993,17 +993,15 @@ void CompactObj::GetString(char* dest) const {
     } else if (taglen_ == SMALL_TAG) {
       size_t decoded_len = DecodedLen(u_.small_str.size());
 
-      // we left some space on the left to allow inplace ascii unpacking.
-      size_t space_left = decoded_len - u_.small_str.size();
       string_view slices[2];
-
       unsigned num = u_.small_str.GetV(slices);
       DCHECK_EQ(2u, num);
-      char* next = dest + space_left;
+      std::string tmp(decoded_len, ' ');
+      char* next = tmp.data();
       memcpy(next, slices[0].data(), slices[0].size());
       next += slices[0].size();
       memcpy(next, slices[1].data(), slices[1].size());
-      detail::ascii_unpack_simd(reinterpret_cast<uint8_t*>(dest + space_left), decoded_len, dest);
+      detail::ascii_unpack_simd(reinterpret_cast<uint8_t*>(tmp.data()), decoded_len, dest);
```

Comment:
I need to check in other places as well. The problem is that we overshoot the buffer. We read starting from `dest + space_left` and we simd load 16 bytes. On the second iteration we increment dest by 14, which causes the starting address to be at pos 18 (because space_left == 4) and we read another 16 bytes overshooting the 32 available bytes